crit-go/scripts/magic_gen.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"github.com/thoas/go-funk"
+	"os"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+func main() {
+
+	if len(os.Args) != 3 {
+		fmt.Println("Usage: magic-gen.py path/to/image.h path/to/magic.json")
+		os.Exit(1)
+	}
+	magic_c_header := os.Args[1]
+	magic_json := os.Args[2]
+	// opens magic file
+	infile, err := os.Open(magic_c_header)
+	check(err)
+	defer infile.Close()
+	outfile, err := os.OpenFile(magic_json, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	check(err)
+	defer outfile.Close()
+	mapper := make(map[string]interface{})
+	all_magic := make(map[string]string)
+	magic := make(map[string]string)
+	by_name := make(map[string]interface{})
+	by_val := make(map[uint64]interface{})
+	scanner := bufio.NewScanner(infile)
+	scanner.Split(bufio.ScanLines)
+
+	var txtlines []string
+
+	for scanner.Scan() {
+		txtlines = append(txtlines, scanner.Text())
+	}
+
+	for _, eachline := range txtlines {
+		split := strings.Fields(eachline)
+		if len(split) < 3 {
+			continue
+		}
+
+		if itemExists(split, "#define") == false {
+			continue
+		}
+
+		key := split[1]
+		value := split[2]
+		//fmt.Println(reflect.TypeOf(value))
+		if funk.Contains(all_magic, value) == true {
+			value := all_magic[value]
+			all_magic[key] = value
+		} else {
+			magic[key] = value
+			all_magic[key] = value
+		}
+	}
+
+	for k, v := range magic {
+		//fmt.Println(reflect.TypeOf(v))
+		if v == "0x0" || v == "1" || k == "0x0" || v == "1" {
+			continue
+		}
+		if strings.Contains(k, "_MAGIC") == true {
+			k = strings.Replace(k, "_MAGIC", "", -1)
+		}
+		vhexedv := hex2int(v)
+		check(err)
+		//fmt.Println(vhexedv)
+		by_name[k] = vhexedv
+		by_val[vhexedv] = k
+	}
+	mapper["by_name"] = by_name
+	mapper["by_val"] = by_val
+	mapjsoned, err := json.Marshal(mapper)
+	check(err)
+	_, err = fmt.Fprintln(outfile, string(mapjsoned))
+	check(err)
+}
+
+func check(e error) {
+	if e != nil {
+		panic(e)
+	}
+}
+
+// function check if item exists in map
+func itemExists(slice interface{}, item interface{}) bool {
+
+	s := reflect.ValueOf(slice)
+	if s.Kind() != reflect.Slice {
+		panic("Invalid data-type")
+	}
+
+	if s.Index(0).Interface() == item {
+		return true
+	} else {
+		return false
+	}
+}
+
+// function to convert hex string to int
+func hex2int(hexStr string) uint64 {
+	// remove 0x suffix if found in the input string
+	cleaned := strings.Replace(hexStr, "0x", "", -1)
```

Comment:
strconv.ParseUint() is perfectly capable of parsing a string with `0x` prefix, see https://golang.org/pkg/strconv/#ParseInt

Reply:
tried that,returns 0 stating invalid no. Thus have to clean the prefix first then send to strconv.ParseUint